Forgot Password Function

Mango-Goose opened the pull request:
Existing users can now change their passwords when they have forgotten it through email.

Mango-Goose committed:
Forgot Password Page

To let users put in their email and send out a reset password email

Mango-Goose committed:
Forgot Password Functionality

Users can now change their password and reset password links can be sent out via email

Mango-Goose committed:
Merge branch 'main' into Naomi

Mango-Goose committed:
Reset Password Confirmation Sorted

Mango-Goose committed:
Merge branch 'main' into Forgot-password

Mango-Goose committed:
Improved formatting of Password Reset email

Mango-Goose committed:
Merge branch 'main' into Forgot-password

Mango-Goose committed:
Welcome Email

Sends out a welcome email when user registers an account

EsamKhalid commented:
![Screenshot 2025-03-06 115044](https://github.com/user-attachments/assets/e2cf68cc-093f-420c-a83c-ec3f0a237178)

EsamKhalid commented:
This pull request includes several changes to the `UserController`, email functionalities, and related views and configurations. The main focus is on implementing a forgot password feature, sending welcome emails, and updating email configurations.

### UserController Enhancements:
* Added imports for `ForgotPassword`, `Welcome`, `Mail`, `Str`, `DB`, and `Log` to `UserController.php`.
* Implemented `forgotPassword` and `resetPassword` methods in `UserController.php` to handle password reset requests and updates.
* Added functionality to send a welcome email upon user registration.

### Email Functionalities:
* Created `ForgotPassword`, `UserVerification`, and `Welcome` mailable classes to handle different types of emails. [[1]](diffhunk://#diff-efc49c864cf09693b0ab859684e48aad545efa5f563e6966efd7c362f82ad2a7R1-R61) [[2]](diffhunk://#diff-6580cae5cdc180de227949b083a426c4e9d0c8a6c754919859e15f097169a02eR1-R53) [[3]](diffhunk://#diff-11b75e0e9af03f6c5f99f64cbeedd1b8bc1728b2ade39958e611bf5c7bd1908dR1-R70)
* Updated the default mailer to use `smtp` and changed the default sender email and name in `mail.php`. [[1]](diffhunk://#diff-af6ac21d51218ff8f16692c0c8a9bf2fb420d3cc0de12458049dd1ed9c5ec244L17-R17) [[2]](diffhunk://#diff-af6ac21d51218ff8f16692c0c8a9bf2fb420d3cc0de12458049dd1ed9c5ec244L112-R113)

### Database and Views:
* Added a migration to create the `password_reset` table.
* Created views for `forgot-password`, `change-password`, and email layouts, including the corresponding Blade templates for the emails. [[1]](diffhunk://#diff-094e5be73a6e41aef732ae991b5c34da733cec9d571d513ee8c6a40ff3eeb017R1-R21) [[2]](diffhunk://#diff-7b474cbc38482e78ab8214ec239adb703c5c758c13663f954609dcd6ddf41b3aR1-R15) [[3]](diffhunk://#diff-79e66364c2668399e0a9caf2773a15015ffc8904d2de708a6481388dfe208419R1-R11) [[4]](diffhunk://#diff-98b44c99cb36a00649d4dd19ffd27d875e3e2b303844ffaa19cc6df88b238c7eR1-R38) [[5]](diffhunk://#diff-39f4c74eeef6e514e87c57ec72bde6a09f471ec1fe364e8007d78934f2230690R1-R10)
* Updated `login.blade.php` to include a "Forgot Password?" link.

### Routes:
* Added routes for forgot password and change password functionalities in `web.php`.

Mango-Goose committed:
Order Confirmation Email

Sends out a simple order confirmation email. Email contains a button to go to the user's order history. Also adjusted email layout

Mango-Goose committed:
Updated Order Confirmation Email Formatting
